xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/Macro.java

```diff
@@ -70,4 +71,16 @@ public interface Macro<P> extends Comparable<Macro<?>>
      */
     List<Block> execute(P parameters, String content, MacroTransformationContext context)
         throws MacroExecutionException;
+
+    /**
+     * Prepare a {@link MacroBlock} meant to be cached to be executed several times. The goal is to pre-execute
+     * everything that can fit in any context and store it in an annotation of the passed {@link MacroBlock}.
```

Comment:
```suggestion
     * everything that is independent of any context and store it in an annotation of the passed {@link MacroBlock}.
```

Maybe clearer, not sure.

Reply:
Yes, it's better.